fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/physical/PruneShuffleColumnRule.java

```diff
@@ -141,6 +152,10 @@ private void prune(DistributionContext childContext) {
                 }
             }
 
+            if (duplicatedColumns && maxColumnIndex == -1) {
+                maxColumnIndex = 0;
+            }
+
```

Comment:
you will choose a bad column when not exists statistic, or choose a low cardinality column